rules_poetry/poetry.bzl

```
@@ -124,7 +169,23 @@ def _impl(repository_ctx):
         if requested.lower() in excludes:
             fail("pyproject.toml dependency {} is also in the excludes list".format(requested))
 
+    toml_markers = {}
+
+    if metadata["lock-version"] in ["2.0"]:
+        poetry_deps = mapping['pyproject']['tool']['poetry']
+        
+        for dep in poetry_deps['dependencies']:
+            extract_markers(repository_ctx, toml_markers, dep, poetry_deps['dependencies'][dep])
+                    
+        if 'group' in poetry_deps:
+            for group in poetry_deps['group']:
+                if 'dependencies' in poetry_deps['group'][group]:
+                    for dep in poetry_deps['group'][group]['dependencies']:
+                        extract_markers(repository_ctx, toml_markers, dep, poetry_deps['group'][group]['dependencies'][dep])
```

It'll aid readability to condense these statements as follows:

```suggestion
        for _, group in poetry_deps.get('group', {}).items():
            for dep, markers in group.get('dependencies', {}).items():
                extract_markers(repository_ctx, toml_markers, dep, markers)
```
The second arg is what is returned if the key does not exists. items method lets you iterate over keys and values at the same time.

---

made these item() usage changes [7aa2a09](https://github.com/mongodb-forks/rules_poetry/pull/1/commits/7aa2a09847cc9482b937f4381021a9cb45946821)